python/pyarrow/tests/test_flight.py

```diff
@@ -877,14 +879,18 @@ def test_repr():
     descriptor_repr = "<pyarrow.flight.FlightDescriptor cmd=b'foo'>"
     endpoint_repr = ("<pyarrow.flight.FlightEndpoint "
                      "ticket=<pyarrow.flight.Ticket ticket=b'foo'> "
-                     "locations=[]>")
+                     "locations=[] "
+                     "expiration_time=2023-04-05 12:34:56+00:00 "
+                     "app_metadata='656e64706f696e7420617070206d65746164617461'>")
```

Comment:
This could read (which is more readable)

    app_metadata=b'endpoint app metadata'

if we wouldn't stick to the C++ implementation of `ToString` here.